crates/task-impls/src/quorum_vote.rs

```diff
@@ -70,12 +74,170 @@ struct VoteDependencyHandle<TYPES: NodeType, I: NodeImplementation<TYPES>> {
     view_number: TYPES::Time,
     /// Event sender.
     sender: Sender<Arc<HotShotEvent<TYPES>>>,
+    /// The current version of HotShot
+    version: Version,
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static> VoteDependencyHandle<TYPES, I> {
+    /// Updates the shared consensus state with the new voting data.
+    async fn update_shared_state(
+        &self,
+        proposal: &QuorumProposal<TYPES>,
+        proposed_leaf: &Leaf<TYPES>,
+        vid_share: &Proposal<TYPES, VidDisperseShare<TYPES>>,
+    ) -> Result<()> {
+        let consensus_reader = self.consensus.read().await;
+        let justify_qc = &proposal.justify_qc;
+
+        // Justify qc's leaf commitment should be the same as the parent's leaf commitment.
+        let parent = consensus_reader
+            .saved_leaves()
+            .get(&justify_qc.date().leaf_commit)
+            .cloned()
+            .context(format!(
+                "Proposal's parent missing from storage with commitment: {:?}, proposal view {:?}",
+                justify_qc.date().leaf_commit,
+                proposal.view_number,
+            ))?;
+
+        let (Some(parent_state), _) = consensus_reader.state_and_delta(parent.view_number()) else {
+            bail!("Parent state not found! Consensus internally inconsistent")
+        };
+
+        drop(consensus_reader);
+
+        let (validated_state, state_delta) = parent_state
+            .validate_and_apply_header(
+                &self.instance_state,
+                &parent,
+                &proposal.block_header.clone(),
+                vid_share.data.common.clone(),
+                self.version,
+            )
+            .await
+            .context("Block header doesn't extend the proposal!")?;
+
+        let state = Arc::new(validated_state);
+        let delta = Arc::new(state_delta);
+
+        // TODO Upgrades
```

Comment:
This block isn't for upgrades, this is validating the DA cert, we're still doing that elsewhere right (in the as a dependency maybe?

Reply:
Yeah Keyao already took care of this, removed.